sonicwall_firewall/assets/logs/sonicwall-firewall.yaml

```diff
@@ -0,0 +1,1090 @@
+      grok:
+        supportRules: ""
+        matchRules: >-
+          source_parser_rule_1
+          :(%{number:network.client.port})?(:)?(%{word:source.network_interface})?(:)?(%{hostname:source.resolved_name})?
+
+          source_parser_rule_2 (%{ip:network.client.ip})?(:)?(%{number:network.client.port})?(:)?(%{word:source.network_interface})?(:)?(%{hostname:source.resolved_name})?
+    - type: grok-parser
+      name: Parsing Sonicwall Firewall nat source
+      enabled: true
+      source: natSrc
+      samples:
+        - 10.10.10.10:48245
+        - 10.10.10.10
+      grok:
+        supportRules: ""
+        matchRules: natscr_parser_rule %{ip:source.nat.ip}(:%{number:source.nat.port})?
+    - type: grok-parser
+      name: Parsing Sonicwall Firewall nat destination
+      enabled: true
+      source: natDst
+      samples:
+        - 10.10.10.10
+        - 10.10.10.10:320
+      grok:
+        supportRules: ""
+        matchRules: natdst_parser_rule
+          %{ip:destination.nat.ip}(:%{number:destination.nat.port})?
+    - type: grok-parser
+      name: Parsing Sonicwall Firewall protocol and service
+      enabled: true
+      source: proto
+      samples:
+        - tcp/http
+        - icmp
+        - ipv6-icmp
+        - udp/netbios-ns
+      grok:
+        supportRules: protocol_service_parser %{regex("[a-zA-Z0-9-]*")}
+        matchRules: proto_parser_rule
+          %{protocol_service_parser:protocol}(/%{protocol_service_parser:service})?
+    - type: grok-parser
+      name: Parsing Sonicwall Firewall time
+      enabled: true
+      source: time
+      samples:
+        - 2024-09-11 06:30:00 UTC
+      grok:
+        supportRules: ""
+        matchRules: time_parser_rule %{date("yyyy-MM-dd HH:mm:ss z"):timestamp}
+    - type: date-remapper
+      name: Define `timestamp` as the official date of the log
+      enabled: true
+      sources:
+        - timestamp
+    - type: grok-parser
+      name: Parsing Sonicwall Firewall message
+      enabled: true
+      source: msg
+      samples:
+        - Assigned IP address 0.0.0.0 to MAC address 00:00:00:00:0C:0E
+        - SYN-Flooding machine %s blacklisted
+        - TCP-Flooding machine %s blacklisted
+        - Possible ARP attack from MAC address %s
+        - RST-Flooding machine %s blacklisted
+      grok:
+        supportRules: ""
+        matchRules: >-
+          message_parser_rule_1 Assigned IP address %{ip:ip_address} to MAC
+          address %{mac:MacAddress}
+
+          message_parser_rule_2 (TCP|SYN|RST|FIN)-Flooding machine %{greedyData:message_value} blacklisted
+
+          message_parser_rule_3 Possible ARP attack from MAC address %{greedyData:message_value}
+
+          message_parser_rule_4 Possible replay attack with this client IP - %{greedyData:message_value}
+    - type: attribute-remapper
+      name: Map `app` to `syslog_appid`
+      enabled: true
+      sources:
+        - app
+      sourceType: attribute
+      target: syslog_appid
```

Comment:
We have mapped it to **syslog_appid** since in 'app' field we get a numeric value instead of name. For better readability we have mapped it to syslog_appid instead of syslog.appname